release-process.md

```diff
@@ -1,54 +1,131 @@
-### 4. Generate release notes
-All of our projects contain a file for release notes in the project
-root directory.  This file is named ``RELEASE.rst``.  For each release it is
-updated with a section titled "Release <version-number>" containing a list
-of the commits in the release.  
-### 5. Update version numbers
-Project version numbers are typically stored in ``settings.py`` and/or
-``setup.py`` depending on the application's architecture and purpose.
-### 6. Commit updates and push ``release-candidate`` branch
-### 7. Generate release notes with checkboxes
-### 8. Open PR to merge ``release-candidate`` branch into ``release`` branch
-### 9. Merge PR once developers verify their commits (manual step)
-### 10. Tag the release
-### 11. Merge the ``release`` branch into the ``master/main`` branch and push
-### 12. Send email notifications (manual step)
-
-
-## Release Version Numbers
-Our releases are identified by a version number that follows the
+  - Code starts out in a pull request which is reviewed by other team members (see [Code Review](code-review.md)).
+  - Then the pull request is merged to the `main` branch (or `master` for legacy projects).
+
+### Process
+
+```mermaid
+sequenceDiagram
+  actor eng as Software Engineers
+  participant branchmain as main/master
+
+  participant branchrc as release-candidate
+  participant envrc as Release Candidate
+
+  participant branchrelease as release
+  participant envprod as Production
+
+  loop
+    eng->>branchmain: Merge pull requests
+  end
+
+  Note over eng: Requests a release candidate
+
+  critical New Release Candidate
+    branchmain->>branchrc: Latest code merged
+    branchrc->>branchrc: Release notes and PR generated
+    branchrc->>envrc: Deploy to RC
+    eng->>envrc: Engineer verifies their commit(s)
+    note over eng: Checks off boxes on the PR
+  end
+
+  critical Production Release
+    eng->>branchrc: Engineer coordinated release approval
+    branchrc->>branchmain: Merged back to main
+    branchmain->>branchrelease: Main branch merged to release
+    branchrelease->>envprod: Deploy to Production
+  end
+
+
+```
+
+#### CI
+
+  - Deployments to the CI environment are only applicable to web applications.
+  - After tests pass on the `master`/`main` branch this triggers a deployment to a CI (continuous integration) server for web application projects.
+
+#### Release candidate
+  
+  - On a daily basis each project is checked on an automated schedule for new changes:
+    - If there are changes a release candidate is created in the `release-candidate` branch.
+    - A new commit with updated release notes and a version bump is made on that branch.
+    - A PR is created on GitHub to merge these changes into `release`.
+    - **Web Applications only:** Once tests pass a deployment to the RC (release candidate) server for web application projects.
+  - Unit tests are run against the release candidate and team members check off checkboxes to verify,
```

Comment:
Which part is? The checkboxes? I think there's still utility in that as both a reminder to engineers what they need to check and an affirmative indication they've done so.

Reply:
Yes, the checkboxes. As with my above comment, I don't have a motive, just making sure that we are deliberate in the decisions that we make here, rather than just cargo-culting behaviors.